build.sh

```diff
@@ -83,26 +83,26 @@ for ARG in "$@"; do
       FILTERED_ALL_PACKAGES=()
       
       for target in "${PACKAGES_TO_BUILD[@]}"; do
-        if containsElement "${target}" "${PACKAGES[@]}"; then
+        if [[ ${#PACKAGES[@]} > 0 && $(containsElement "${target}" "${PACKAGES[@]}"; echo $?) == 0 ]]; then
           FILTERED_PACKAGES+=(${target})
         fi
-        if containsElement "${target}" "${TSC_PACKAGES[@]}"; then
+        if [[ ${#TSC_PACKAGES[@]} > 0 && $(containsElement "${target}" "${TSC_PACKAGES[@]}"; echo $?) == 0 ]]; then
           FILTERED_TSC_PACKAGES+=(${target})
         fi
-        if containsElement "${target}" "${NODE_PACKAGES[@]}"; then
+        if [[ ${#NODE_PACKAGES[@]} > 0 && $(containsElement "${target}" "${NODE_PACKAGES[@]}"; echo $?) == 0 ]]; then
           FILTERED_NODE_PACKAGES+=(${target})
         fi
-        if containsElement "${target}" "${ALL_PACKAGES[@]}"; then
+        if [[ ${#ALL_PACKAGES[@]} > 0 && $(containsElement "${target}" "${ALL_PACKAGES[@]}"; echo $?) == 0 ]]; then
           FILTERED_ALL_PACKAGES+=(${target})
         fi
       done
-      PACKAGES=("${FILTERED_PACKAGES[@]}")
-      TSC_PACKAGES=("${FILTERED_TSC_PACKAGES[@]}")
-      NODE_PACKAGES=("${FILTERED_NODE_PACKAGES[@]}")
-      ALL_PACKAGES=("${FILTERED_ALL_PACKAGES[@]}")
+      [[ ${#FILTERED_PACKAGES} -gt 0 ]] && PACKAGES=("${FILTERED_PACKAGES[@]}") || PACKAGES=()
```

Comment:
I think we should use a method or the other but we shouldn't mix both.
We should do this (as in PowerShell):
```bash
[[ ${#FILTERED_PACKAGES} -gt 0 ]]
[[ ${#FILTERED_PACKAGES} -eq 0 ]]
...
```

Or this (as in javascript and other languages):
```bash
[[ ${#FILTERED_PACKAGES} > 0 ]]
[[ ${#FILTERED_PACKAGES} == 0 ]]
...
```

Could you please adapt this in the entire script ? 😊 

We should check which one is the most compliant (on Linux, Windows, MacOS)